### DISCUSSION TITLE
Order of properties when writing

### DISCUSSION THREAD

**Discussion by jkaiser-ioki** [bjornharrtell/jts2geojson#51] [Q&A] [answered]
Hi,

we are using jts2geojson to serialise some features and feature collections, but ran into problems when doing some read/write consistency round-trip tests, that involved comparing two geojson files as text files. The main issue was that order of key-value-pairs in the GeoJSON feature `properties` field got shuffled around. As far as I know this is well within the accepted specs of JSON which does not demand any kind of ordering of dictionaries , but there's also the possibility to do stuff like this to establish some form of canonical ordering:

```
ObjectMapper mapper = new ObjectMapper();
mapper.setConfig(mapper.getSerializationConfig()
   .with(MapperFeature.SORT_PROPERTIES_ALPHABETICALLY));
   .with(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS)
```

This of course relies on having access to the `mapper` object, which at the moment I believe is a private/inaccessible member of your `GeoJSONFactory` class, or am I overlooking something? I.e., is it at the moment possible to enforce any output order of the properties on a read-write cycle, either preserving the input order or fixing to alphabetical somehow, that I am not seeing? If it is relevant: we are currently using your version 0.16.1, on java 11

**Comment by bjornharrtell** [comment]
In my opinion it is wrong to do text comparison of json for round trip tests, you should be using some json aware comparison because as you are aware of the properties is not an ordered set.

That said, I don't know if it's possible to control in any way and this project is more or less in almost non existent maintenance so you will have to investigate this on your own.

Sorry for perhaps a disappointing answer.

**Comment by jkaiser-ioki** [comment, marked as answer]
Hi, thank you for the quick response. I partially agree with your assessment that JSON file comparisons are not ideal and was just hoping for a potentially quick solution.

the fact that this project is essentially unmaintained is also valuable info.

So all in all not disappointing, as I can now focus looking elsewhere.

So thanks again :-)
